internal/pkg/api/handleCheckin.go

```diff
@@ -498,13 +510,12 @@ func (ct *CheckinT) writeResponse(zlog zerolog.Logger, w http.ResponseWriter, r
 	if len(payload) > compressThreshold && compressionLevel != flate.NoCompression && acceptsEncoding(r, kEncodingGzip) {
 		wrCounter := datacounter.NewWriterCounter(w)
 
-		zipper, err := gzip.NewWriterLevel(wrCounter, compressionLevel)
-		if err != nil {
-			return fmt.Errorf("writeResponse new gzip: %w", err)
-		}
+		zipper, _ := ct.gwPool.Get().(*gzip.Writer)
```

Comment:
Add a comment that the effectiveness of the pool depends on spreading these responses out over time.

Reply:
This comment should probably explicitly reference the importance of the ActionThrottle configuration parameter.

Reply:
added in struct defintion